spec/rubocop/cop/style/single_argument_dig_spec.rb

```diff
@@ -13,6 +13,17 @@
           { key: 'value' }[:key]
         RUBY
       end
+
+      it 'registers an offense and corrects unsuitable use of dig with safe navigation operator' do
+        expect_offense(<<~RUBY)
+          { key: 'value' }&.dig(:key)
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `{ key: 'value' }[:key]` instead of `{ key: 'value' }&.dig(:key)`.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          { key: 'value' }[:key]
```

Comment:
@koic 

I think there is an error here.
```ruby
hsh = { key: 'value' }
hsh&.dig(:key) # => 'value'
hsh[:key] # => 'value'

hsh = nil
hsh&.dig(:key) # => nil
hsh[:key] # => NoMethodError (undefined method `[]' for nil:NilClass)
```

I think there is the following:
```ruby
expect_correction(<<~RUBY)
  { key: 'value' }&.fetch(:key, nil)
RUBY
```

Reply:
Sure, I've opened #12519 to revert this change.